src/functions.php

```diff
@@ -272,9 +314,15 @@ function is_last_pipeline_operator_write(array $pipeline): bool
         return false;
     }
 
-    $lastOp = (array) $lastOp;
+    if (! is_array($lastOp) && ! is_object($lastOp)) {
+        return false;
+    }
+
+    $lastOp = document_to_array($lastOp);
+
+    reset($lastOp);
 
-    return in_array(key($lastOp), ['$out', '$merge'], true);
+    return key($lastOp) === '$merge' || key($lastOp) === '$out';
```

Comment:
We should use rector to track that automatically. https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#php73

Reply:
> Consider using [array_key_first](https://www.php.net/manual/en/function.array-key-first.php) (available as of PHP 7.3.0, symfony/polyfill might be required) instead of resetting the array pointer. This likely applies to a number of other usages as well.

TIL those functions exist. Will do!

> We should use rector to track that automatically. https://github.com/rectorphp/rector/blob/main/docs/rector_rules_overview.md#php73

@GromNaN: Would you mind creating a PHPLIB ticket to propose adding Rector to our CI?

Reply:
Of course: https://jira.mongodb.org/browse/PHPLIB-1141